### ISSUE TITLE
ArangoSearch: using 'ANY IN' syntax results in 'ArangoSearch noncompliant expression' error

### ISSUE THREAD

**Issue by oliverlockwood** [arangodb/arangodb#11054]
## My Environment

* __ArangoDB Version__:        3.4.7
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   Docker
* __Configuration__:               3 nodes
* __Infrastructure__:               AWS
* __Operating System__:        Alpine
* __Total RAM in your machine__:        Varies
* __Disks in use__:        Varies
* __Used Package__:              Debian

## Component, Query & Data

__Affected feature__:
ArangoSearch

__AQL query (if applicable)__:
```
FOR object IN view_our_objects
  SEARCH object.visibility ANY IN [ 1, 2, 123 ]
  LIMIT 10
  RETURN { 'object' : object }
```

__AQL explain (if applicable)__:
```
xecution plan:
 Id   NodeType            Site  Est.   Comment
  1   SingletonNode       DBS      1   * ROOT
  2   EnumerateViewNode   DBS      1     - FOR object IN view_flex_objects SEARCH (object.`visibility` any in [ 1, 2, 123 ])   /* view query */
  4   CalculationNode     DBS      1       - LET #1 = { "object" : object }   /* simple expression */
  8   RemoteNode          COOR     1       - REMOTE
  9   GatherNode          COOR     1       - GATHER 
  3   LimitNode           COOR     1       - LIMIT 0, 10
  5   ReturnNode          COOR     1       - RETURN #1

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   handle-arangosearch-views
  3   scatter-arangosearch-view-in-cluster
  4   distribute-filtercalc-to-cluster
  5   remove-unnecessary-remote-scatter
```

__Dataset__:
Example object in collection backing the `view_our_objects` view:
```
{
  "name": "redacted",
  "visibility": [
    4,
    123,
    456
  ]
}
```

## Steps to reproduce

1. Run the query

__Problem__:
The Arango Java driver throws an exception with the following stacktrace:
```

com.arangodb.ArangoDBException: Response: 501, Error: 9 - AQL: Error message received from shard '' on cluster node 'PRMR-915867f6-effc-4646-b542-8f78d2ffa376': Error message received from shard '' on cluster node 'CRDN-5ce81711-c9a3-467a-af37-856c0a2b275a': Error message received from shard '' on cluster node 'PRMR-886f1d29-6bae-4c0f-a42e-aa1db2664f7d': not implemented: Unable to evaluate loop variable 'object' as a part of ArangoSearch noncompliant expression '$0.visibility any array IN [4,10202]' (while executing)
	at com.arangodb.internal.util.ResponseUtils.checkError(ResponseUtils.java:53)
	at com.arangodb.internal.velocystream.VstCommunication.checkError(VstCommunication.java:149)
	at com.arangodb.internal.velocystream.VstCommunicationSync.execute(VstCommunicationSync.java:128)
	at com.arangodb.internal.velocystream.VstCommunicationSync.execute(VstCommunicationSync.java:42)
	at com.arangodb.internal.velocystream.VstCommunication.execute(VstCommunication.java:132)
	at com.arangodb.internal.velocystream.VstProtocol.execute(VstProtocol.java:47)
	at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:79)
	at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:59)
	at com.arangodb.internal.ArangoDatabaseImpl.query(ArangoDatabaseImpl.java:198)
```

__Expected result__:
The query returns as normal.

__Hypothesis__:
My suspicion is that the `ANY IN` syntax (as documented [here](https://www.arangodb.com/docs/stable/aql/operators.html#array-comparison-operators), for comparing an array to an array) is not supported in ArangoSearch.

From experimentation, it looks to me like I can achieve the desired behaviour by just using `IN` instead.

Is that correct?  If not, what would you suggest?

Perhaps there could be a section of the documentation for ArangoSearch-compatible AQL?  At present, it seems to be an undocumented subset, which is a bit of a pain to develop against.

Many thanks.

**Comment by Simran-B** [comment]
Support for array comparison operators in SEARCH expressions was added to ArangoSearch just recently in v3.6.0:

https://www.arangodb.com/docs/stable/release-notes-new-features36.html#dynamic-search-expressions-with-arrays

Please upgrade to use this feature.

**Comment by Simran-B** [comment]
Oh, and the document attribute reference needs to be on the right-hand side, so:

```js
FOR object IN view_our_objects
  SEARCH [ 1, 2, 123 ] ANY IN object.visibility
  LIMIT 10
  RETURN { 'object' : object }
```

**Comment by maxkernbach** [comment]
Hi @oliverlockwood,

As already mentioned, support for array comparison operators in SEARCH expressions is available in 3.6.

Did you have a chance to upgrade to latest 3.6 and run the modfied query as suggested by Simran?

**Comment by maxkernbach** [comment]
Hi @oliverlockwood,

Since we have not heard back from you recently, I am closing this issue for now.
Feel free to reopen this ticket if needed.
